test/validator_test.exs

```diff
@@ -108,6 +108,11 @@ defmodule ValidatorTest do
     assert {:error, "Required property tag was not present.", "#/pet"} =
              Validator.validate("/post/pets", %{"id" => 1, "pet" => %{"name" => "pet_name"}})
 
+    # assert {:error, "Type mismatch. Expected String but got Null.", "#/pet/nickname"} =
```

Comment:
Before modifying the validator, this was the result of using a 'x-nullable field'. I added the test to confirm how it previously worked. Then I update the validator, and wrote the test showing the desired behavior. I can remove the comment, but I wanted to show the previous behavior to make this PR more understandable.